assets/queries/k8s/seccomp_profile_is_not_configured/query.rego

```diff
@@ -1,199 +1,91 @@
-		"keyActualValue": "'spec.jobTemplate.spec.template.metadata.annotations.seccomp.security.alpha.kubernetes.io/defaultProfileName' is undefined",
+		"keyExpectedValue": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.seccompProfile.type should be set to 'RuntimeDefault' or 'Localhost'", [metadata.name, specInfo.path, containerType, container.name]),
+		"keyActualValue": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.seccompProfile.type is set to '%s'", [metadata.name, specInfo.path, containerType, container.name, profile]),
 	}
 }
 
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") == "CronJob"
-
-	metadata := document.spec.jobTemplate.spec.template.metadata
-
-	parentMetadata := document.metadata
-	common_lib.valid_key(metadata, "annotations")
-	annotations := metadata.annotations
+# pod defines seccompProfile.type and container inherits this setting
+checkSeccompProfile(specInfo, container, containerType, document, metadata) = result {
+	containerCtx := object.get(container.securityContext, "seccompProfile", {})
+	not common_lib.valid_key(containerCtx, "type")
 
-	common_lib.valid_key(annotations, "seccomp.security.alpha.kubernetes.io/defaultProfileName")
-
-	seccomp := annotations["seccomp.security.alpha.kubernetes.io/defaultProfileName"]
-
-	seccomp != "runtime/default"
+	profile := specInfo.spec.securityContext.seccompProfile.type
+	not any([profile == "RuntimeDefault", profile == "Localhost"])
 
 	result := {
-		"documentId": input.document[i].id,
-		"searchKey": sprintf("metadata.name={{%s}}.spec.jobTemplate.spec.template.metadata.annotations.seccomp.security.alpha.kubernetes.io/defaultProfileName", [parentMetadata.name]),
+		"documentId": document.id,
+		"searchKey": sprintf("metadata.name={{%s}}.%s.securityContext.seccompProfile.type=%s", [metadata.name, specInfo.path, profile]),
 		"issueType": "IncorrectValue",
-		"keyExpectedValue": "'spec.jobTemplate.spec.template.metadata.annotations.seccomp.security.alpha.kubernetes.io/defaultProfileName' is 'runtime/default'",
-		"keyActualValue": sprintf("'spec.jobTemplate.spec.template.metadata.annotations.seccomp.security.alpha.kubernetes.io/defaultProfileName' is '%s'", [seccomp]),
+		"keyExpectedValue": sprintf("metadata.name={{%s}}.%s.securityContext.seccompProfile.type should be set to 'RuntimeDefault' or 'Localhost'", [metadata.name, specInfo.path]),
+		"keyActualValue": sprintf("metadata.name={{%s}}.%s.securityContext.seccompProfile.type is set to '%s'", [metadata.name, specInfo.path, profile]),
 	}
 }
 
-###################################################################
+# neither pod nor container define seccompProfile.type
+checkSeccompProfile(specInfo, container, containerType, document, metadata) = result {
+	specCtx := object.get(specInfo.spec, "securityContext", {})
+	specSeccompCtx := object.get(specCtx, "seccompProfile", {})
+	not common_lib.valid_key(specSeccompCtx, "type")
```

Comment:
Please note that these checks were added on purpose like this:
`specInfo.spec.securityContext` may or may not exist (and the rule handles this accordingly). The same applies to `specInfo.spec.securityContext.seccompProfile`. 

The alternative you suggested would let the rule evaluate to _false_ in case `specInfo.spec.securityContext.seccompProfile` is not defined. However, at this execution path, this would be an expectable behavior.